### DISCUSSION TITLE
Calling function without api call and handle default argument

### DISCUSSION THREAD

**Discussion by yeyeric** [fastapi/fastapi#8605] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Body

app = FastAPI()

@app.post('/my_function')
def my_function(my_bool: bool = Body(default=False, embed=True)):
    if my_bool:
        print('my_bool is true')
    else:
        print('my_bool is false')


@app.post('/another_function_calling_my_function')
def another_function_calling_my_function():
    my_function()
```

### Description

Hello,

calling my_function using an api call and sending (or not) my_bool as param behaves as expected (ie if not sent, my_bool is False, if sent, then my_bool is the value_sent)

However, I sometimes need to call the same function my_function without api call. However, if I don't explicitly pass "my_bool" as an argument, it will be set to Body(default=False) which is considered as boolean True by python.

How should I do without explicitly adding the argument when my_function is called to have this evaluated to its default_value ?

Thanks

### Operating System

Linux, Windows

### FastAPI Version

0.74.1

### Python Version

3.9

**Comment by jgould22** [comment, marked as answer]
You should probably just extract whatever code it is into its own function or module and keep your api code (the function warped in the decorator) separate.

You can then just call it from your end points and keep it out of the parsing and validation logic which will help with unit tests as well.

```python
from fastapi import FastAPI, Body

app = FastAPI()

def extracted_function(param: bool = False):
    if param:
        print('param is true')
    else:
        print('param is false')

@app.post('/my_function')
def my_function(my_bool: bool = Body(default=False, embed=True)):
    extracted_function(my_bool)

@app.post('/another_function_calling_my_function')
def another_function_calling_my_function():
    extracted_function()
```
